internal/proxyservice/proxy_service.go

```diff
@@ -1,43 +1,147 @@
 package proxyservice
 
 import (
+	"net"
+	"strings"
+
+	"github.com/cyberark/secretless-broker/internal"
 	"github.com/cyberark/secretless-broker/internal/plugin"
 	v1 "github.com/cyberark/secretless-broker/internal/plugin/v1"
+	tcpproxy "github.com/cyberark/secretless-broker/internal/proxyservice/tcp"
 	"github.com/cyberark/secretless-broker/pkg/secretless"
 	v2 "github.com/cyberark/secretless-broker/pkg/secretless/config/v2"
-	"github.com/cyberark/secretless-broker/pkg/secretless/log"
+	logapi "github.com/cyberark/secretless-broker/pkg/secretless/log"
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector"
+	"github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector/tcp"
 )
 
 // TODO: move to impl package
 type proxyServices struct {
-	config        v2.Config
-	logger        log.Logger
-	eventNotifier v1.EventNotifier
-	availPlugins  plugin.AvailablePlugins
+	config          v2.Config
+	logger          logapi.Logger
+	eventNotifier   v1.EventNotifier
+	availPlugins    plugin.AvailablePlugins
+	runningServices []secretless.Service
 }
 
-// TODO: Rename to Call or Run and return a Stopper instead of having Stop()
-// Start starts all proxy services
-func (s *proxyServices) Start() {
-	// TODO: Implement
+type connectorResources struct {
+	logger logapi.Logger
+	config []byte
+}
+
+func (cr *connectorResources) Logger() logapi.Logger {
+	return cr.logger
+}
+
+func (cr *connectorResources) Config() []byte {
+	return cr.config
+}
+
+func newConnectorResources(l logapi.Logger, cfg []byte) connector.Resources {
+	return &connectorResources{
+		logger: l,
+		config: cfg,
+	}
+}
 
-	// For each ProxyService:
-	//
-	// 1. Rewrap the Logger with service name prefix
-	// 2. Create the ConnectorResources object
-	// etc...
+// Start starts all proxy services
+func (s *proxyServices) Start() error {
+	for _, svc := range s.servicesToStart() {
+		err := svc.Start()
+		if err != nil {
+			s.logger.Errorf("could not start proxy service: %s", err)
+			continue
+		}
+		s.runningServices = append(s.runningServices, svc)
+	}
+	return nil
 }
 
 // Stop stops all proxy services
-func (s *proxyServices) Stop() {
-	// Stop all the Service
+func (s *proxyServices) Stop() error {
+	for _, svc := range s.runningServices {
+		err := svc.Stop()
+		if err != nil {
+			s.logger.Errorf("could not stop proxy service: %s", err)
+		}
+	}
+	return nil
+}
+
+func (s *proxyServices) servicesToStart() []secretless.Service {
+	var servicesToStart []secretless.Service
+
+	tcpPlugins := s.availPlugins.TCPPlugins()
+	// httpPlugins := s.availPlugins.HTTPPlugins()
+
+	for _, svc := range s.config.Services {
+		requestedPlugin := svc.Connector //TODO: this rename is a name smell
+
+		// first check the available TCP Plugins
+		tcpPlugin, found := tcpPlugins[requestedPlugin]
+		if found {
+			if tcpSvc := s.createTCPService(svc, tcpPlugin); tcpSvc != nil {
```

Comment:
This probably should return an `error` instead of us relying on the `nil` as the error condtion

Reply:
Error here should probably be a fatal error

Reply:
yeah, i'll just go ahead and make this fatal.